test/suite/test_prefetch02.py

```diff
@@ -0,0 +1,143 @@
+                            session_cfg='prefetch=(enabled=true)', prefetch=True)),
+        ('config_c', dict(conn_cfg='prefetch=(available=false,default=false),statistics=(all)', 
+                            session_cfg='', prefetch=False)),
+    ]
+
+    prefetch_scenarios = [
+        ('forward-traversal', dict(prefetch_scenario='forward-traversal', scenario_type='traversal')),
+        ('backward-traversal', dict(prefetch_scenario='backward-traversal', scenario_type='traversal')),
+        ('verify', dict(prefetch_scenario='verify', scenario_type='verify')),
+    ]
+
+    scenarios = make_scenarios(config_options, prefetch_scenarios)
+
+    def get_stat(self, stat, session_name):
+        stat_cursor = session_name.open_cursor('statistics:')
+        val = stat_cursor[stat][2]
+        stat_cursor.close()
+        return val
+
+    def get_prefetch_activity_stats(self, session_name):
+        pages_queued = self.get_stat(wiredtiger.stat.conn.block_prefetch_pages_queued, session_name)
+        prefetch_attempts = self.get_stat(wiredtiger.stat.conn.block_prefetch_attempts, session_name)
+        prefetch_pages_read = self.get_stat(wiredtiger.stat.conn.block_prefetch_pages_read, session_name)
+        return pages_queued, prefetch_attempts, prefetch_pages_read
+
+    def get_prefetch_skipped_stat(self, session_name):
+        prefetch_skips = self.get_stat(wiredtiger.stat.conn.block_prefetch_skipped, session_name)
+        return prefetch_skips
+
+    # Checks for pre-fetching activity by asserting that relevant statistics have increased.
+    def check_prefetching_activity(self, session_name, pages_queued, prefetch_attempts, prefetch_pages_read):
+        new_pages_queued, new_prefetch_attempts, new_prefetch_pages_read = self.get_prefetch_activity_stats(session_name)
+        self.assertGreaterEqual(new_pages_queued, pages_queued)
+        self.assertGreater(new_prefetch_attempts, prefetch_attempts)
+        self.assertGreaterEqual(new_prefetch_pages_read, prefetch_pages_read)
+
+    # Checks that the values of statistics related to pre-fetching activity are equal to zero,
+    # and that pages are being skipped when deciding whether to pre-fetch the page or not.
+    def check_no_prefetching_activity(self, session_name, prefetch_skips):
+        new_prefetch_skips = self.get_prefetch_skipped_stat(session_name)
+        pages_queued, prefetch_attempts, prefetch_pages_read = self.get_prefetch_activity_stats(session_name)
+        self.assertGreaterEqual(new_prefetch_skips, prefetch_skips)
+        self.assertEqual(pages_queued, 0)
+        self.assertEqual(prefetch_attempts, 0)
+        self.assertEqual(prefetch_pages_read, 0)
+
+    def test_prefetch_scenarios(self):
+        os.mkdir(self.new_dir)
+        helper.copy_wiredtiger_home(self, '.', self.new_dir)
+
+        new_conn = self.wiredtiger_open(self.new_dir, self.conn_cfg)
+        s = new_conn.open_session(self.session_cfg)
+        s.create(self.uri, 'key_format=i,value_format=i')
```

Comment:
It seems like this could be parameterised (via `make_scenarios`) relatively easily?